src/db/spanner/manager.rs

```diff
@@ -72,18 +72,23 @@ impl<T: std::marker::Send + std::marker::Sync + 'static> ManageConnection
     type Error = grpcio::Error;
 
     async fn connect(&self) -> Result<Self::Connection, Self::Error> {
-        // Requires GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account.json
-        let creds = ChannelCredentials::google_default_credentials()?;
-
-        // Create a Spanner client.
-        let chan = ChannelBuilder::new(self.env.clone())
-            .max_send_message_len(100 << 20)
-            .max_receive_message_len(100 << 20)
-            .secure_connect(SPANNER_ADDRESS, creds);
+        let chan = {
+            // Requires
+            // GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account.json
+            // XXX: issue732: Could google_default_credentials (or
+            // ChannelBuilder::secure_connect) block?!
+            let creds = ChannelCredentials::google_default_credentials()?;
```

Comment:
Note the added nesting here is to drop `ChannelCredentials` (which is not `Send`) before the await call. Otherwise we end up with the issue described in detail in this [blog post](https://blog.rust-lang.org/inside-rust/2019/10/11/AsyncAwait-Not-Send-Error-Improvements.html).

```
error: future cannot be sent between threads safely
  --> src/db/spanner/manager.rs:74:70
   |
74 |       async fn connect(&self) -> Result<Self::Connection, Self::Error> {
   |  ______________________________________________________________________^
75 | |         // Requires GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account.json
76 | |         // XXX: async?
77 | |         let creds = ChannelCredentials::google_default_credentials()?;
...  |
94 | |         })
95 | |     }
   | |_____^ future returned by `__connect` is not `Send`
   |
   = help: within `impl core::future::future::Future`, the trait `std::marker::Send` is not implemented for `*mut grpcio_sys::bindings::grpc_channel_credentials`
note: future is not `Send` as this value is used across an await
  --> src/db/spanner/manager.rs:88:23
   |
77 |         let creds = ChannelCredentials::google_default_credentials()?;
   |             ----- has type `grpcio::security::credentials::ChannelCredentials` which is not `Send`
...
88 |         let session = create_session(&client, &self.database_name).await?;
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ await occurs here, with `creds` maybe used later
...
95 |     }
   |     - `creds` is later dropped here
   = note: required for the cast to the object type `dyn core::future::future::Future<Output = std::result::Result<db::spanner::manager::SpannerSession, grpcio::error::Error>> + std::marker::Send`

error: aborting due to previous error
```